bin/improver-ecc

```diff
@@ -0,0 +1,154 @@
+    Copula Coupling.
+    """
+    parser = ArgParser(
+        description='Apply Ensemble Copula Coupling '
+                    'to a file with one cube.')
+
+    # General options:
+    parser.add_argument('input_filepath', metavar='INPUT_FILE',
+                        help='A path to an input NetCDF file to be processed')
+    parser.add_argument('output_filepath', metavar='OUTPUT_FILE',
+                        help='The output path for the processed NetCDF')
+    parser.add_argument('--no_of_percentiles', default=None, type=int,
+                        metavar='NUMBER_OF_PERCENTILES',
+                        help='The number of percentiles to be generated.')
+    parser.add_argument('--sampling_method', default='quantile',
+                        const='quantile', nargs='?',
+                        choices=['quantile', 'random'],
+                        metavar='PERCENTILE_SAMPLING_METHOD',
+                        help='Method to be used for generating the list of '
+                             'percentiles with forecasts generated at each '
+                             'percentile. "quantile" is the default option.')
+
+    # Different use cases:
+    # (We can either reorder OR rebadge)
+    group = parser.add_mutually_exclusive_group(required=True)
+
+    group.add_argument('--reordering', default=False, action='store_true',
+        help='The option used to create ensemble members '
+             'from percentiles by reordering the input '
+             'percentiles based on the order of the '
+             'raw ensemble forecast.')
+    group.add_argument('--rebadging', default=False, action='store_true',
+        help='The option used to create ensemble members '
+              'from percentiles by rebadging the input '
+              'percentiles.')
+
+    # If reordering, can do so either based on original members, or randomly.
+    reordering = parser.add_argument_group(
+        'Reordering options', 'Options for reordering the input percentiles '
+        'using the raw ensemble forecast as required to create ensemble '
+        'members.')
+    reordering.add_argument('--raw_forecast_filepath',
+                        metavar='RAW_FORECAST_FILE',
+                        help='A path to an raw forecast NetCDF file to be '
+                             'processed.')
+    reordering.add_argument('--random_ordering', default=False,
+                        action='store_true',
+                        help='Decide whether or not to use random ordering '
+                             'within the ensemble reordering step.')
+    reordering.add_argument('--fixed_random_seed', default=None,
+                        help='Decide whether or not a fixed random seed is '
+                             'used to create the random numbers, either '
+                             'used for the random_ordering or for splitting '
+                             'tied values.')
+
```

Comment:
I think "fixed" is redundant here, and might actually be misleading. In many pseudo-random number generators, the word "seed" is used not only to refer to the initial seed, but also for the input used to produce the next number along the sequence (i.e: the "seed" changes due to the state of the RNG).